deriv_api/deriv_api.py

```diff
@@ -1,48 +1,123 @@
+from deriv_api.cache import Cache
 from deriv_api.deriv_api_calls import DerivAPICalls
+from deriv_api.in_memory import InMemory
 import websockets
 import json
 import logging
+from deriv_api.errors import APIError, ConstructionError
+from deriv_api.utils import dict_to_cache_key, is_valid_url
+import re
 
 # TODO: remove after development
 logging.basicConfig(
     format="%(asctime)s %(message)s",
-    level=logging.DEBUG
+    level=logging.ERROR
 )
+
 class DerivAPI(DerivAPICalls):
-    """Main class of the python DerivAPI module. It provides methods to connect, read and interact with API"""
+    """
+    The minimum functionality provided by DerivAPI, provides direct calls to the API.
+    `api.cache` is available if you want to use the cached data
+
+    example
+    apiFromEndpoint = deriv_api.DerivAPI({ endpoint: 'ws.binaryws.com', app_id: 1234 });
+
+    param {Object}     options
+    param {WebSocket}  options.connection - A ready to use connection
+    param {String}     options.endpoint   - API server to connect to
+    param {Number}     options.app_id     - Application ID of the API user
+    param {String}     options.lang       - Language of the API communication
+    param {String}     options.brand      - Brand name
+    param {Object}     options.middleware - A middleware to call on certain API actions
+
+    property {Cache} cache - Temporary cache default to {InMemory}
+    property {Cache} storage - If specified, uses a more persistent cache (local storage, etc.)
+    """
     wsconnection:str = ''
+    storage = ''
+    def __init__(self, options):
```

Comment:
Maybe 
```suggestion
    def __init__(self, **options):
```
will be better ?